### PR TITLE
BUG: fix edit paren behavior

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/ast.py [hunk 1]
```diff
@@ -2285,42 +2285,78 @@ class Edit(Oper):
...
+                pass
+            else:
+                yield newn, path.runt.initPath(newn)
+
+    async def run(self, runt, genr):
+
+        form = runt.snap.model.forms.get(self.name)
         if form is None:
-            raise s_exc.NoSuchForm(name=name)
+            raise s_exc.NoSuchForm(name=self.name)
```
Comment (vEpiphyte) [inline synapse/lib/ast.py]: This is duplicate of the work done in prepare()

File: synapse/tests/test_lib_ast.py [hunk 2]
```diff
@@ -286,6 +292,27 @@ class AstTest(s_test.SynTest):
             errs = [m[1] for m in mesgs if m[0] == 'err']
             self.eq(errs[0][0], 'BadPropValu')
 
+    async def test_ast_editparen_fromnodes(self):
+
+        async with self.getTestCore() as core:
+
+            nodes = await core.nodes('[ inet:ipv4=1.2.3.4 ]  [ (inet:dns:a=(vertex.link, $node.value()) +#foo ) ]')
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_ast.py]: just add this to the earlier test

File: synapse/lib/ast.py (hunk 1 above)
Comment (invisig0th) [reply synapse/lib/ast.py]: Yep, thanks!  Moved it to prepare and forgot to cull

File: synapse/lib/ast.py [hunk 3]
```diff
@@ -2347,42 +2347,81 @@ class Edit(Oper):
...
 class EditNodeAdd(Edit):
 
+    def prepare(self):
+
+        oper = self.kids[1].value()
+        self.name = self.kids[0].value()
+
+        self.form = self.core.model.form(self.name)
+        if self.form is None:
+            raise s_exc.NoSuchForm(name=self.name)
```
Comment (vEpiphyte) [inline synapse/lib/ast.py]: test this case :)

**Pull request merged**